presidio-tester/cmd/presidio-tester/tester.go

```diff
@@ -0,0 +1,196 @@
+		Body:   ioutil.NopCloser(bytes.NewReader(payload)),
+		Header: http.Header{
+			"Content-Type": []string{"application/json"},
+		},
+	}
+
+	resp, err := http.DefaultClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+	if resp.StatusCode >= 300 {
+		log.Error("%s %s: expected status code smaller than 300 , got '%d'\n", request.Method, request.URL.String(), resp.StatusCode)
+		panic("Bad status code")
+	}
+	defer resp.Body.Close()
+	bodyBytes, _ := ioutil.ReadAll(resp.Body)
+	return string(bodyBytes)
+}
+
+func invokeHTTPUpload(path string, values map[string]io.Reader) []byte {
+
+	var b bytes.Buffer
+	w := multipart.NewWriter(&b)
+	for key, r := range values {
+		var fw io.Writer
+		var err error
+		if x, ok := r.(io.Closer); ok {
+			defer x.Close()
+		}
+		// Add an image file
+		if x, ok := r.(*os.File); ok {
+			fw, err = w.CreateFormFile(key, x.Name())
+
+		} else {
+			// Add other fields
+			fw, err = w.CreateFormField(key)
+		}
+		if err != nil {
+			panic(err)
+		}
+		_, err = io.Copy(fw, r)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	w.Close()
+
+	req, err := http.NewRequest("POST", "http://"+APIEP+path, &b)
+	if err != nil {
+		panic(err)
+	}
+	req.Header.Set("Content-Type", w.FormDataContentType())
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	if res.StatusCode != http.StatusOK {
+		log.Error("bad status: %s", res.Status)
+		panic("Bad status code")
+	}
+	defer res.Body.Close()
+	bodyBytes, _ := ioutil.ReadAll(res.Body)
+	return bodyBytes
+}
+
+// TestAddTemplate Test Add Template
+func TestAddTemplate() {
+	log.Info("Start TestAddTemplate")
+	log.Info("Adding analyze template")
+	payload := generatePayload("analyze-template.json")
+	invokeHTTPRequest("/api/v1/templates/test/analyze/test", "POST", payload)
+
+	log.Info("Adding anonymize template")
+	payload = generatePayload("anonymize-template.json")
+	invokeHTTPRequest("/api/v1/templates/test/anonymize/test", "POST", payload)
+	log.Info("End TestAddTemplate")
+}
+
+//TestDeleteTemplate Test Delete Template
+func TestDeleteTemplate() {
+	log.Info("Start TestDeleteTemplate")
+	log.Info("Delete analyze template")
+	invokeHTTPRequest("/api/v1/templates/test/analyze/test", "DELETE", []byte(""))
+	log.Info("Delete anonymize template")
+	invokeHTTPRequest("/api/v1/templates/test/anonymize/test", "DELETE", []byte(""))
+	log.Info("End TestDeleteTemplate")
+}
+
+//TestAnalyzer Test Analyzer
+func TestAnalyzer() {
+	log.Info("Start TestAnalyzer")
+	TestAddTemplate()
+	payload := generatePayload("analyze-request.json")
+	log.Info("Analyze request")
+	invokeHTTPRequest("/api/v1/projects/test/analyze", "POST", payload)
+	log.Info("End TestAnalyzer")
+}
+
+//TestAnonymizer Test Anonymizer
+func TestAnonymizer() {
+	log.Info("Start TestAnonymizer")
+	TestAddTemplate()
+	payload := generatePayload("anonymize-request.json")
+	log.Info("anonymize request")
+	invokeHTTPRequest("/api/v1/projects/test/anonymize", "POST", payload)
```

Comment:
where is the validation of the results? other than 200 OK

Reply:
At the moment I am using the existing e2e test, not creating new ones. We can add a task to make the e2e test more robust.

Reply:
Ok I have added a little bit more logic to the test to avoid only http 200 OK.
It now also validates the actual responses as well.